docs/upgrade.md

```diff
@@ -168,6 +169,13 @@ In Marionette 3, the HTML will be:
 </div>
 ```
 
+### Arguments for Lifecycle Events
+
+The arguments to all lifecycle events were completely changed. All events will
```

Comment:
Kind of.  Most events already received it.  What changed was that all events were made consistent.

I believe all of the breaking events because of this change were (each having a `before:`):
- Object `destroy`
- Region `show`
- View's region add and remove events
- CollectionView's add and remove child events
- Application `start`

Every other event was either already passed `this` as the first arg, or had no args to begin with.